cargo apple build fails since it cannot find built static libraries

When running the build for any iOS device, including `*-apple-ios-sim` targets, the compiled static libraries are compiled to `$CARGO_MANIFEST_DIR/target`, which if you then run `cargo apple build` results in XCode not being able to find your compiled static library since it's expecting it to be in `$CARGO_MANIFEST_DIR/gen/apple/Sources`.

Simply copying the compiled target-arch folder, e.g `aarch64-apple-ios` from `target/aarch64-apple-ios` to `gen/apple/Sources/aarch64-apple-ios` will let XCode compile correctly and then you can run `cargo apple archive` to generate an installer to publish on your dev devices (Or run `cargo apple dev` to run it in the iOS sim device instead of).

Sym linking the folder from `target` to `gen/apple/Sources` didn't seem to work either, so it seems like you either need to copy the generated target arch folder each time you want to build, or override the generated cargo build output directory with `CARGO_TARGET_DIR=gen/apple/Sources cargo build --target *-apple-ios[-sim]`.

Would be nice to have this automatically copied across or set as the build output directory when running `cargo apple build`.

Note: This only affects `cargo apple build`, building from XCode, e.g to run on iOS sim, will look in the `target` folder, which is confusing.

Hello @DanielRWhite and @wusyong  running `cargo apple run` fails and I believe it is related to this issue.
I don't have much experience on how to solve this, can you point out what I need to do to add the library to the correct folder?

The logs 
```
cargo apple run

=== BUILD TARGET tauri-example-wry_iOS OF PROJECT tauri-example-wry WITH CONFIGURATION debug ===
ld: warning: directory not found for option '-L/Users/username/workspace/tauri-example-wry/gen/apple/Sources/aarch64-apple-ios/debug'
ld: library not found for -ltauri_example_wry
clang: error: linker command failed with exit code 1 (use -v to see invocation)
** BUILD FAILED **

error: Failed to build via `xcodebuild`
    command ["xcodebuild"] exited with code 65
```

I tried
[Unable to get IOS working on Mac M1](https://github.com/tauri-apps/tauri-mobile/issues/81#issuecomment-1419571051)  but it did not work.

--- a/.changes/xcode-script.md
+++ b/.changes/xcode-script.md
@@ -1,0 +1,5 @@
+---
+"tauri-mobile": patch
+---
+
+Add xcode script back and skip it when building simulator target.

--- a/src/apple/cli.rs
+++ b/src/apple/cli.rs
@@ -443,9 +443,17 @@ impl Exec for Input {
                 let isysroot = format!("-isysroot {}", sdk_root.display());
 
                 for arch in arches {
+                    // FIXME Build the rust crate for iOS Simulator target too.
+                    if arch == "Simulator" {
+                        continue;
+                    }
+
                     // Set target-specific flags
                     let (triple, rust_triple) = match arch.as_str() {
                         "arm64" => ("aarch64_apple_ios", "aarch64-apple-ios"),
+                        // FIXME triple for cflags seems incorrect and we don't actually need to
+                        // set it when cross compile simulator target.
+                        // "arm64-sim" => ("aarch64_apple_ios", "aarch64-apple-ios"),
                         "x86_64" => ("x86_64_apple_ios", "x86_64-apple-ios"),
                         _ => return Err(Error::ArchInvalid { arch }),
                     };

--- a/templates/platforms/xcode/project.yml.hbs
+++ b/templates/platforms/xcode/project.yml.hbs
@@ -112,6 +112,12 @@ targets:
         basedOnDependencyAnalysis: {{this.based-on-dependency-analysis}}{{/if}}{{#if this.discovered-dependency-file}}
         discoveredDependencyFile: {{this.discovered-dependency-file}}{{/if}}
       {{~/each}}
+      - script: ${HOME}/.cargo/bin/cargo-apple xcode-script -v --platform ${PLATFORM_DISPLAY_NAME:?} --sdk-root ${SDKROOT:?} --framework-search-paths "${FRAMEWORK_SEARCH_PATHS:?}" --header-search-paths "${HEADER_SEARCH_PATHS:?}" --gcc-preprocessor-definitions "${GCC_PREPROCESSOR_DEFINITIONS:?}" --configuration ${CONFIGURATION:?} ${FORCE_COLOR} ${ARCHS:?} ${FEATURES}
+        name: Build Rust Code
+        basedOnDependencyAnalysis: false
+        outputFiles:
+          - $(SRCROOT)/target/aarch64-apple-ios/${CONFIGURATION}/deps/lib{{snake-case app.name}}.a
+          - $(SRCROOT)/target/x86_64-apple-ios/${CONFIGURATION}/deps/lib{{snake-case app.name}}.a
     {{~#if ios-post-compile-scripts}}
     postCompileScripts:
       {{~#each ios-post-compile-scripts}}{{#if this.path}}

